File: mobile-flutter/lib/api/services.dart
import 'package:dio/dio.dart';
import 'api_client.dart';

class AuthService {
  final Dio _dio = apiClient.dio;

  Future<Map<String, dynamic>> login(String email, String password) async {
    final response = await _dio.post('/auth/sign-in', data: {
      'email': email,
      'password': password,
    });
    return response.data;
  }

  Future<Map<String, dynamic>> logout() async {
    final response = await _dio.post('/auth/sign-out');
    return response.data;
  }

  Future<Map<String, dynamic>> getProfile() async {
    final response = await _dio.get('/auth/profile');
    return response.data;
  }
}

class DashboardService {
  final Dio _dio = apiClient.dio;

  Future<Map<String, dynamic>> getStats() async {
    final response = await _dio.get('/dashboard');
    return response.data;
  }
}

class AttendanceService {
  final Dio _dio = apiClient.dio;

  Future<Map<String, dynamic>> clockIn() async {
    final response = await _dio.post('/clock-in');
    return response.data;
  }

  Future<Map<String, dynamic>> clockOut() async {
    final response = await _dio.post('/clock-out');
    return response.data;
  }

  Future<Map<String, dynamic>> getWorkLogs({String? startDate, String? endDate}) async {
    final response = await _dio.get('/work-logs', queryParameters: {
      if (startDate != null) 'start_date': startDate,
      if (endDate != null) 'end_date': endDate,
    });
    return response.data;
  }
}

class LeaveService {
  final Dio _dio = apiClient.dio;

  Future<Map<String, dynamic>> getCategories() async {
    final response = await _dio.get('/time-off-categories');
    return response.data;
  }

  Future<Map<String, dynamic>> getRequests({String? status}) async {
    final response = await _dio.get('/time-off-requests', queryParameters: {
      if (status != null) 'status': status,
    });
    return response.data;
  }

  Future<Map<String, dynamic>> createRequest({
    required int timeOffCategoryId,
    required String startDate,
    required String endDate,
    required String reason,
  }) async {
    final response = await _dio.post('/time-off-requests', data: {
      'time_off_category_id': timeOffCategoryId,
      'start_date': startDate,
      'end_date': endDate,
      'reason': reason,
    });
    return response.data;
  }

  Future<Map<String, dynamic>> getBalance() async {
    final response = await _dio.get('/time-off-balance');
    return response.data;
  }
}

final authService = AuthService();
final dashboardService = DashboardService();
final attendanceService = AttendanceService();
final leaveService = LeaveService();

File: mobile-expo/src/api/services.ts
import { api } from './client';
import { ApiResponse, AuthResponse, DashboardStats, TimeOffRequest, TimeOffCategory, WorkLog, User } from '../types';

export const authService = {
  login: async (email: string, password: string): Promise<AuthResponse> => {
    const response = await api.post('/auth/sign-in', { email, password });
    return response.data;
  },

  logout: async (): Promise<ApiResponse<null>> => {
    const response = await api.post('/auth/sign-out');
    return response.data;
  },

  getProfile: async (): Promise<ApiResponse<{ user: User }>> => {
    const response = await api.get('/auth/profile');
    return response.data;
  },
};

export const dashboardService = {
  getStats: async (): Promise<ApiResponse<DashboardStats>> => {
    const response = await api.get('/dashboard');
    return response.data;
  },
};

export const attendanceService = {
  clockIn: async (): Promise<ApiResponse<WorkLog>> => {
    const response = await api.post('/clock-in');
    return response.data;
  },

  clockOut: async (): Promise<ApiResponse<WorkLog>> => {
    const response = await api.post('/clock-out');
    return response.data;
  },

  getWorkLogs: async (params?: { start_date?: string; end_date?: string }): Promise<ApiResponse<WorkLog[]>> => {
    const response = await api.get('/work-logs', { params });
    return response.data;
  },

  getSummary: async (params: { staff_member_id: number; start_date: string; end_date: string }): Promise<ApiResponse<any>> => {
    const response = await api.get('/attendance-summary', { params });
    return response.data;
  },
};

export const leaveService = {
  getCategories: async (): Promise<ApiResponse<TimeOffCategory[]>> => {
    const response = await api.get('/time-off-categories');
    return response.data;
  },

  getRequests: async (params?: { status?: string }): Promise<ApiResponse<TimeOffRequest[]>> => {
    const response = await api.get('/time-off-requests', { params });
    return response.data;
  },

  createRequest: async (data: {
    time_off_category_id: number;
    start_date: string;
    end_date: string;
    reason: string;
  }): Promise<ApiResponse<TimeOffRequest>> => {
    const response = await api.post('/time-off-requests', data);
    return response.data;
  },

  getBalance: async (): Promise<ApiResponse<any>> => {
    const response = await api.get('/time-off-balance');
    return response.data;
  },
};
